loopchain/blockchain/votes/v0_1a/votes.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2018 ICON Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import Counter
+from typing import Iterable, List, Dict
+from loopchain.blockchain.types import Hash32, ExternalAddress
+from loopchain.blockchain.votes import Votes as BaseVotes
+from loopchain.blockchain.votes.v0_1a import BlockVote, LeaderVote
+
+
+class BlockVotes(BaseVotes[BlockVote]):
+    VoteType = BlockVote
+
+    def __init__(self, reps: Iterable['ExternalAddress'], voting_ratio: float,
+                 block_height: int, block_hash: Hash32):
+        self.block_height = block_height
+        self.block_hash = block_hash
+        super().__init__(reps, voting_ratio)
+
+    def verify(self, vote: BlockVote):
+        super().verify(vote)
+
+        if vote.block_height != self.block_height:
+            raise RuntimeError(f"Vote block_height not match. {vote.block_height} != {self.block_height}\n"
+                               f"{vote}")
+
+        if vote.block_hash != self.block_hash and vote.block_hash != Hash32.empty():
+            raise RuntimeError(f"Vote block_hash not match. {vote.block_hash} != {self.block_hash}\n"
+                               f"{vote}")
+
+    def empty_vote(self, rep: ExternalAddress):
+        return self.VoteType.empty(rep, self.block_height)
+
+    def is_completed(self):
+        return self.get_result() is not None
+
+    def get_result(self):
+        true_vote_count = sum(1 for vote in self.votes
+                              if not self.is_empty_vote(vote) and vote.block_hash == self.block_hash)
+        if true_vote_count >= self.quorum:
+            return True
+
+        false_vote_count = sum(1 for vote in self.votes
+                               if not self.is_empty_vote(vote) and vote.block_hash == Hash32.empty())
+        if false_vote_count >= len(self.reps) - self.quorum + 1:
+            return False
+        return None
+
+    def __eq__(self, other: 'BlockVotes'):
+        return \
+            super().__eq__(other) and \
+            self.block_hash == other.block_hash and \
+            self.block_height == other.block_height
```

Comment:
I suggest parenthesis instead of back slash.
```
    return (
        super().__eq__(other) and
        self.block_hash == other.block_hash and
        self.block_height == other.block_height)
```